### PR TITLE
Move (most of) No.Swiftmodule into a proper API test.

### PR DESCRIPTION
**Pull request by adrian-prantl**
Testing debugging from just reflection metadata is very important on
all platforms. Shell tests only run on the host. By migrating this
test to an API test we can make sure this is also tested when
debugging on iOS devices.

### CODE REVIEW

File: lldb/packages/Python/lldbsuite/test/make/Makefile.rules [hunk 1]
```diff
@@ -599,18 +599,22 @@ VPATHSOURCES=$(patsubst %,$(VPATH)/%,$(SWIFT_SOURCES)) $(patsubst %,$(VPATH)/%,$
...
-	  $(patsubst -g,,$(SWIFTFLAGS)) -o "$(EXE)"
+	$(SWIFTC) $(LD_EXTRAS) $(OBJECTS) $(patsubst -g,,$(SWIFTFLAGS)) -o "$(EXE)"
 ifneq "$(CODESIGN)" ""
 	$(CODESIGN) -s - "$(EXE)"
 endif
 else # OS = Linux
 $(EXE): $(MODULENAME).swiftmodule.o $(OBJECTS)
 	@echo "### Linking" $(EXE)
-ifneq "$(EXCLUDE_WRAPPED_SWIFTMODULE)" ""
+ifneq "$(HIDE_SWIFTMODULE)" ""
```
Comment (adrian-prantl) [inline lldb/packages/Python/lldbsuite/test/make/Makefile.rules:618]: Turns out this was no longer used by any test...

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by kastiglione — APPROVED** [review]

File: lldb/test/API/lang/swift/reflection_only/lib.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+public enum NoPayload {
+  case first
+  case second
+}
+
+public enum WithPayload {
+  case empty
+  case with(i: Int)
+}
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/reflection_only/lib.swift:32]: Just a thought: should we include a multi-payload enum here, commented out, with a link to the radar?

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Pull request merged**